addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -364,6 +364,12 @@ export class ProductScreen extends Component {
                 "pos_categ_ids",
                 this.pos.selectedCategory.id
             );
+            if(this.pos.selectedCategory.child_id){
+                const child_ids = this.pos.selectedCategory.child_id;
+                for (const category of child_ids) {
+                    list = list.concat(this.pos.models["product.product"].getBy("pos_categ_ids", category.id));
```

Comment:
do you think you could use `getAllChildren` from  `point_of_sale/static/src/app/models/pos_category.js` here?